src/mixins.js

```diff
@@ -0,0 +1,20 @@
+import { css } from 'styled-components';
+
+/**
+ * Text mixin: defines typography properties: base font family, custom
+ * font-size and custom color. See theme.js for avaliable values.
+ *
+ * @param {string} [$.family='base'] Font family
+ * @param {string} [$.size='base'] Font size
+ * @param {string} [$.color='base'] Color
+ */
+export const text = ({
```

Comment:
I think at this point we should introduce [styled-system](https://github.com/jxnblk/styled-system) since it's too close to what it offers. This is something I want to do anyways since it's way too helpful IMO.

Reply:
I'm probably missing something about styled-system, but as I see you can only control actual values via component props? And making this kind of mixing isn't possible.

Introducing styled-system may be still a good idea ;-) Do you have some example how do you use it?

Reply:
I'm not sure about what kind of mixing you're talking about but yes, I'm using styled-system on 2 projects already and it's been great. I can show you some examples but they aren'y OSS :(

Reply:
I see now. Yes, generally speaking, by design you can control individual properties using props. The closest to mixing is [complexStyles](https://github.com/jxnblk/styled-system/blob/master/docs/api.md#complexstyle) which are a sub-sets of styles. There are no mixins as you did it here but I think it's actually a good design decision since you might want to give access over one part of typographic props but not over another on a specific component.

Reply:
I don't really see use cases for props like `color`, I think we should use functional prop names instead (like `primary` and `secondary`).

And to defend my mixin, using styled-system directly would look like this:

```
${fontSize({fontSize: 'base'})}
${color({color: 'base'})}
...
```

instead of just:

```
${text()}
```

if you only need defaults.

Of course we can implement the mixin using styled-system functions.

But maybe it's not something very useful in the end ;-)